File: python/allergies.py
MAP = {
    'eggs': 1,
    'peanuts': 2,
    'shellfish': 4,
    'strawberries': 8,
    'tomatoes': 16,
    'chocolate': 32,
    'pollen': 64,
    'cats': 128,
}


class Allergies:

    def __init__(self, score: int):
        self.score = score

    def allergic_to(self, item: str) -> bool:
        return self.score & MAP[item] != 0

    @property
    def lst(self):
        return [k for (k, v) in MAP.items() if v & self.score]

File: lua/allergies.lua
local function list(score)
    local items = {
        "eggs",
        "peanuts",
        "shellfish",
        "strawberries",
        "tomatoes",
        "chocolate",
        "pollen",
        "cats",
    }
    local t = {}
    for i, item in ipairs(items) do
        local n = 2 ^ (i - 1)
        if score & n == n then
            table.insert(t, item)
        end
    end
    return t
end

local function allergic_to(score, which)
    local lst = list(score)
    for _, item in ipairs(lst) do
        if item == which then
            return true
        end
    end
    return false
end

return { list = list, allergic_to = allergic_to }
